### PR TITLE
changelog 24.7.0 

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
### Description

✂️ snip snip ✂️ the making of a new release.

Xref #13995

### CODE REVIEW

File: docs/source/dev-guide/plugins/index.rst [hunk 1]
```diff
@@ -133,7 +133,7 @@ found. These automatic reports can be really verbose and can get in the way of c
 *expected* errors. See `this issue in conda-build`_ as an example.
 
 To mark exceptions as *expected*, plugins should raise :class:`conda.CondaError` or a subclass
-thereof. See [`conda_auth.exceptions`](https://github.com/conda-incubator/conda-auth/blob/0.2.1/conda_auth/exceptions.py) for an example.
+thereof. See `conda_auth.exceptions`_ for an example.
```
Comment (kenodegard) [inline docs/source/dev-guide/plugins/index.rst:136]: ## Before:

Screenshot 2024-07-18 at 13 14 36

## After:

Screenshot 2024-07-18 at 13 15 10

File: conda/gateways/logging.py [hunk 2]
```diff
@@ -183,6 +184,11 @@ def initialize_std_loggers():
     verbose_logger.propagate = False
 
 
+@deprecated("25.3", "25.9", addendum="Unused.")
+def initialize_root_logger(level=ERROR):
+    attach_stderr_handler(level=level, filters=[TokenURLFilter()])
```
Comment (kenodegard) [inline conda/gateways/logging.py:189]: Forgot to mark as deprecated in #13735

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,94 @@
 [//]: # (current developments)
 
+## 24.7.0 (2024-07-17)
+
+### Enhancements
+
+* MSYS2 packages can now use the upstream installation prefixes. (#13649)
```
Comment (kenodegard) [inline CHANGELOG.md:7]: Removing this since it was included in the 24.5.0 release.

```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,94 @@
...
+* Add a new `reporters` setting for configure output. (#13736)
+* Report traceback of plugin loading errors with verbosity 2 or higher (`-vv` or more). (#13742 via #13846)
+* Skip checking for `.pyc` and `.pyo` files in the `conda doctor` "missing files" health check. (#13370 via #13931)
+* **Breaking change**  `conda list --explicit` will not print authentication details by default. A new flag `--auth` has been added so folks can opt-in to this behaviour. (#13936)
+* Print transaction report for `@EXPLICIT` lockfile installs too. (#13940)
+* Do not require `-n/--name` or `-p/--prefix` if `conda create` is invoked with `--dry-run`. (#13941)
+Add an `envvars_force_uppercase` setting which defaults to `True`, uppercasing all environment variables (thereby justifying `conda`'s current behaviour); when `envvars_force_uppercase` is set to `False`, conda will only save preserved-case variable names. (#13713 via #13943)
```
Comment (kenodegard) [inline CHANGELOG.md:14]: Missing bullet point:

```suggestion
* Add an `envvars_force_uppercase` setting which defaults to `True`, uppercasing all environment variables (thereby justifying `conda`'s current behaviour); when `envvars_force_uppercase` is set to `False`, conda will only save preserved-case variable names. (#13713 via #13943)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,94 @@
...
+* Alias `conda env list` command to `conda info --envs`. (#13972)
+
+### Bug fixes
+
+* Improve treatment of logger levels (#13735)
```
Comment (kenodegard) [inline CHANGELOG.md:19]: Missing period:

```suggestion
* Improve treatment of logger levels. (#13735)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,94 @@
...
+* Mask authentication details in `conda-meta/*.json` metadata. (#13937)
+* Mask Anaconda.org tokens in verbose logs. (#13939, #13987)
+* Fixes parsing error when history file only contains a single commented line (#13960)
```
Comment (kenodegard) [inline CHANGELOG.md:22]: Use imperative tense and add missing period:

```suggestion
* Fix parsing error when history file only contains a single commented line. (#13960)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,94 @@
...
+* Add missing `emscripten` and `wasi` entries to the recognized platforms, and `wasm32` to the recognized architectures. (#13095)
+* Checksum comparisons in `conda.gateways.connection.download.download()` are not case sensitive anymore. (#13969)
```
Comment (kenodegard) [inline CHANGELOG.md:24]: Imperative tense:

```suggestion
* Fix checksum comparisons in `conda.gateways.connection.download.download()` to be case insensitive. (#13969)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,94 @@
...
+* Disallow some more characters in Windows for prefix names (`^`, `%`, `!`, `=`, `(`, `)`, `\`). These characters complicate or prevent environment activation if present. (#12558 via #13975)
+* Fix caching when `repodata.json` contains `\r\n` line endings. (#14002 via #14003)
+* Fix binary_replace not matching chunks that end with `\n`. (#14043 via #14044)
```
Comment (kenodegard) [inline CHANGELOG.md:27]:
```suggestion
* Fix `conda.core.portability.binary_replace` not matching chunks that end with `\n`. (#14043 via #14044)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,94 @@
...
+* Mark `conda.utils.msys2_shell_base` as pending deprecation. Use `conda.activate` instead. (#14034)
+* Mark `conda.utils.shells` as pending deprecation. Use `conda.activate` instead. (#14034)
+
+### Docs
+
+* Clarify proxy server configuration in documentation. (#12856)
+* Added type hints and doc strings to `conda.core.envs_manager`. (#13817)
+* Add logging overview as deep-dive (#13735)
+* Update conda cheatsheet text and add it directly to cheatsheet page. (#13889)
+* Added cheatsheet PDF download to cheatsheet page. (#13909)
```
Comment (kenodegard) [inline CHANGELOG.md:57]:
```suggestion
* Add cheatsheet PDF download to cheatsheet page. (#13909)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,94 @@
...
+* Added `ssl_verify: truststore` to the user guide. (#13935)
+* Fixed the help text of the `satisfied-skip-solve` flag. (#13946)
```
Comment (kenodegard) [inline CHANGELOG.md:59]:
```suggestion
* Fix the help text of the `satisfied-skip-solve` flag. (#13946)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,94 @@
...
+* Add a section explaining how to correctly raise exceptions from a plugin. (#13741 via #13950)
+* Add new article for configuring `envs_dirs` and `pkgs_dirs`. (#13954)
+
+### Other
+
+* Replace calls to logger.warn with logger.warning (#13963)
```
Comment (kenodegard) [inline CHANGELOG.md:65]:
```suggestion
* Replace calls to `logger.warn` with `logger.warning`. (#13963)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,93 @@
 [//]: # (current developments)
 
+## 24.7.0 (2024-07-17)
+
+### Enhancements
+
+* Add a new `reporters` setting for configure output. (#13736)
```
Comment (beeankha) [inline CHANGELOG.md:7]:
```suggestion
* Add a new `reporters` setting for configuring output. (#13736)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,93 @@
...
+* Mark `conda.utils.translate_stream` as pending deprecation. (#14034)
+* Mark `conda.utils.unix_shell_base` as pending deprecation. Use `conda.activate` instead. (#14034)
+* Mark `conda.utils.msys2_shell_base` as pending deprecation. Use `conda.activate` instead. (#14034)
+* Mark `conda.utils.shells` as pending deprecation. Use `conda.activate` instead. (#14034)
+
+### Docs
+
+* Clarify proxy server configuration in documentation. (#12856)
+* Add type hints and doc strings to `conda.core.envs_manager`. (#13817)
+* Add logging overview as deep-dive (#13735)
```
Comment (beeankha) [inline CHANGELOG.md:54]:
```suggestion
* Add logging overview as deep-dive. (#13735)
```

**Review by kenodegard — APPROVED** [review]

**Pull request merged**